### ISSUE TITLE
Build failed when upgrading from 0.0.5 to 0.0.8

### ISSUE THREAD

**Issue by duncanmak** [arvyy/kawa-maven-plugin#5]
My project is working fine with 0.0.5, but when I upgrade to 0.0.8, I see this:

```
[INFO] --- kawa-maven-plugin:0.0.8:compile (compile) @ edwin-editor ---
Exception in thread "main" java.lang.Error: attempting to push dead variable
	at gnu.bytecode.CodeAttr.emitLoad(CodeAttr.java:1392)
	at gnu.expr.Declaration.load(Declaration.java:415)
	at gnu.expr.SetExp.compile(SetExp.java:223)
	at gnu.expr.Expression.compileWithPosition(Expression.java:166)
	at gnu.expr.Expression.compileWithPosition(Expression.java:151)
	at gnu.kawa.functions.AppendValues.compile(AppendValues.java:76)
	at gnu.expr.ApplyExp.inlineCompile(ApplyExp.java:832)
	at gnu.expr.ApplyExp.compile(ApplyExp.java:225)
	at gnu.expr.ApplyExp.compile(ApplyExp.java:167)
	at gnu.expr.Expression.compileWithPosition(Expression.java:166)
	at gnu.expr.LambdaExp.compileBody(LambdaExp.java:1661)
	at gnu.expr.Compilation.generateBytecode(Compilation.java:1950)
	at gnu.expr.Compilation.process(Compilation.java:1832)
	at gnu.expr.ModuleInfo.loadEager(ModuleInfo.java:337)
	at kawa.standard.require.importDefinitions(require.java:289)
	at kawa.standard.ImportFromLibrary.handleImport(ImportFromLibrary.java:486)
	at kawa.standard.ImportFromLibrary.scanImportSet1(ImportFromLibrary.java:295)
	at kawa.standard.ImportFromLibrary.scanImportSet(ImportFromLibrary.java:263)
	at kawa.standard.ImportFromLibrary.scanForm(ImportFromLibrary.java:96)
	at kawa.lang.Translator.scanForm(Translator.java:1623)
	at kawa.lang.Translator.scanBody(Translator.java:1680)
	at kawa.standard.begin.scanForm(begin.java:23)
	at kawa.lang.Translator.scanForm(Translator.java:1623)
	at gnu.kawa.lispexpr.LispLanguage.parse(LispLanguage.java:117)
	at gnu.expr.Language.parse(Language.java:765)
	at gnu.expr.Language.parse(Language.java:759)
	at gnu.expr.Language.parse(Language.java:753)
	at kawa.repl.compileFiles(repl.java:768)
	at kawa.repl.processArgs(repl.java:451)
	at kawa.repl.main(repl.java:830)
```

**Comment by arvyy** [comment]
are you by any chance trying to compile some sort of library (as opposed to a final application with a clear `main` entry)?

**Comment by arvyy** [comment]
The ultimate change, was that it was redesigned to only compile single `main` file with `--main` flag (with the other things compiled as needed transitively through imports). In a case of a library and what not, they were changed to not be compiled, but instead have their source packed into an auxiliary artifact. The reason it's done so, is because I realized compiling library with one set of flags makes them incompatible with main application jar if it was compiled with another set of flags. Using libraries in a source form solved this issue
